### PR TITLE
fix an error caused by Abp.Account.EnableLocalLogin not configured

### PR DESCRIPTION
**Pull request by gicexxopode**
fix an error caused by Abp.Account.EnableLocalLogin not configured

#### maliming: **This pr moves `AccountSettingDefinitionProvider` from `Volo.Abp.Account.Application` to `Volo.Abp.Account.Application.Contracts`.**

### CODE REVIEW

**Comment by gicexxopode** [comment]
AuthServer.Host Login 500 Error，config file AccountSettingDefinitionProvider.cs move from Volo.Abp.Account.Application to Volo.Abp.Account.ApplicationContracts

**Comment by maliming** [comment]
hi @gicexxopode 

Try replacing `AbpIdentityApplicationContractsModule` with `AbpIdentityApplicationModule`.

https://github.com/abpframework/abp/blob/e84dde4f76abf5ccb45036afce868bde9266e4f4/samples/MicroserviceDemo/applications/AuthServer.Host/AuthServer.Host.csproj#L28

https://github.com/abpframework/abp/blob/e84dde4f76abf5ccb45036afce868bde9266e4f4/samples/MicroserviceDemo/applications/AuthServer.Host/AuthServerHostModule.cs#L36

```
 <ProjectReference Include="..\..\..\..\modules\identity\src\Volo.Abp.Identity.Application\Volo.Abp.Identity.Application.csproj" />

typeof(AbpIdentityApplicationModule),
```

**Comment by gicexxopode** [comment]
This also works, I just followed the convention and put the settings file into the contracts project

File: modules/account/src/Volo.Abp.Account.Application/Volo.Abp.Account.Application.csproj [hunk 1]
```diff
@@ -16,4 +16,8 @@
     <ProjectReference Include="..\..\..\..\framework\src\Volo.Abp.UI.Navigation\Volo.Abp.UI.Navigation.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Volo\Abp\Account\Settings\" />
+  </ItemGroup>
+
```
Comment (maliming) [inline modules/account/src/Volo.Abp.Account.Application/Volo.Abp.Account.Application.csproj]: You should remove these.

**Comment by gicexxopode** [comment]
Fix page / Account / Manage failure in AuthServer.Host project

**Comment by hikalkan** [comment]
Thanks @gicexxopode.
@maliming you can merge if no problem (I didn't check).

**Comment by maliming** [comment]
@hikalkan I'm not sure which layer we should put `SettingDefinitionProvider `on?

**Comment by yinchang0626** [comment]
@maliming @gicexxopode 
I do a test in this pull request branch (gicexxopode/dev)
the error still exist:
I just follow [doc](https://docs.abp.io/en/abp/latest/Samples/Microservice-Demo) to do.
Is this pull request already resolve this issue?
Is there any others problem?? 
```text
Volo.Abp.AbpException: Undefined setting: Abp.Account.EnableLocalLogin
   at Volo.Abp.Settings.SettingDefinitionManager.Get(String name) in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.Settings\Volo\Abp\Settings\SettingDefinitionManager.cs:line 40
   at Volo.Abp.Settings.SettingProvider.GetOrNullAsync(String name) in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.Settings\Volo\Abp\Settings\SettingProvider.cs:line 26
   at Volo.Abp.Settings.SettingProviderExtensions.IsTrueAsync(ISettingProvider settingProvider, String name) in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.Settings\Volo\Abp\Settings\SettingProviderExtensions.cs:line 14
   at Volo.Abp.Account.Web.Pages.Account.IdentityServerSupportedLoginModel.OnGetAsync() in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\modules\account\src\Volo.Abp.Account.Web.IdentityServer\Pages\Account\IdentityServerSupportedLoginModel.cs:line 82
   at Castle.DynamicProxy.AsyncInterceptorBase.ProceedAsynchronous[TResult](IInvocation invocation, IInvocationProceedInfo proceedInfo)
   at Volo.Abp.Castle.DynamicProxy.CastleAbpMethodInvocationAdapterWithReturnValue`1.ProceedAsync() in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.Castle.Core\Volo\Abp\Castle\DynamicProxy\CastleAbpMethodInvocationAdapterWithReturnValue.cs:line 24
   at Volo.Abp.Uow.UnitOfWorkInterceptor.InterceptAsync(IAbpMethodInvocation invocation) in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.Uow\Volo\Abp\Uow\UnitOfWorkInterceptor.cs:line 25
   at Volo.Abp.Castle.DynamicProxy.CastleAsyncAbpInterceptorAdapter`1.InterceptAsync[TResult](IInvocation invocation, IInvocationProceedInfo proceedInfo, Func`3 proceed) in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.Castle.Core\Volo\Abp\Castle\DynamicProxy\CastleAsyncAbpInterceptorAdapter.cs:line 29
   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Convert[T](Object taskAsObject)
   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.ExecutorFactory.GenericTaskHandlerMethod.Execute(Object receiver, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeHandlerMethodAsync()
   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeNextPageFilterAsync()
   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Rethrow(PageHandlerExecutedContext context)
   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ExceptionContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Volo.Abp.AspNetCore.Auditing.AbpAuditingMiddleware.InvokeAsync(HttpContext context, RequestDelegate next) in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.AspNetCore\Volo\Abp\AspNetCore\Auditing\AbpAuditingMiddleware.cs:line 41
   at Volo.Abp.AspNetCore.Auditing.AbpAuditingMiddleware.InvokeAsync(HttpContext context, RequestDelegate next) in C:\Users\YinChang\Documents\Works\Bitbucket\FS\abp\framework\src\Volo.Abp.AspNetCore\Volo\Abp\AspNetCore\Auditing\AbpAuditingMiddleware.cs:line 46
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass5_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by maliming** [comment]
hi @yinchang0626 Which project?

**Comment by yinchang0626** [comment]
@maliming 
`\samples\MicroserviceDemo\applications\AuthServer.Host`
 test in this pull request.

**Comment by maliming** [comment]
Yes, `AuthServer.Host` should reference xx`Volo.Abp.Account.Application`.

@gicexxopode Can you test it?

File: samples/MicroserviceDemo/applications/AuthServer.Host/AuthServer.Host.csproj [hunk 2]
```diff
@@ -25,6 +25,7 @@
     <ProjectReference Include="..\..\..\..\framework\src\Volo.Abp.EntityFrameworkCore.SqlServer\Volo.Abp.EntityFrameworkCore.SqlServer.csproj" />
     <ProjectReference Include="..\..\..\..\framework\src\Volo.Abp.AspNetCore.Mvc.UI.Theme.Basic\Volo.Abp.AspNetCore.Mvc.UI.Theme.Basic.csproj" />
     <ProjectReference Include="..\..\..\..\framework\src\Volo.Abp.EventBus.RabbitMQ\Volo.Abp.EventBus.RabbitMQ.csproj" />
+    <ProjectReference Include="..\..\..\..\modules\identity\src\Volo.Abp.Identity.Application\Volo.Abp.Identity.Application.csproj" />
```
Comment (maliming) [inline samples/MicroserviceDemo/applications/AuthServer.Host/AuthServer.Host.csproj]: `AuthServer.Host` should reference `Volo.Abp.Account.Application`

**Comment by abllyboy** [comment]
I pulled the 2.0.1 now,the error also exists.I added reference Volo.Abp.Account.Application,modify the Module and replace AbpIdentityApplicationContractsModule to AbpIdentityApplicationModule in the AuthServer.Host,but no use,

**Comment by abllyboy** [comment]
I resovled it,to deal this should reference Volo.Abp.Account.Application,and add its module "AbpAccountApplicationModule" in the AuthServer.Host's module 
like below
```
 <ProjectReference Include="..\..\..\..\modules\account\src\Volo.Abp.Account.Application\Volo.Abp.Account.Application.csproj" />

[DependsOn(
        typeof(AbpAutofacModule),
        typeof(AbpEventBusRabbitMqModule),
        typeof(AbpPermissionManagementEntityFrameworkCoreModule),
        typeof(AbpAuditLoggingEntityFrameworkCoreModule),
        typeof(AbpSettingManagementEntityFrameworkCoreModule),
        typeof(AbpIdentityEntityFrameworkCoreModule),
        typeof(AbpIdentityApplicationModule),
        typeof(AbpAccountApplicationModule),
       // typeof(AbpIdentityApplicationContractsModule),
        typeof(AbpIdentityServerEntityFrameworkCoreModule),
        typeof(AbpEntityFrameworkCoreSqlServerModule),
        typeof(AbpAccountWebIdentityServerModule),
        typeof(AbpAspNetCoreMvcUiBasicThemeModule)
        )]
    public class AuthServerHostModule : AbpModule
    {
...

**Pull request closed without merging**